### DISCUSSION TITLE
Trouble with ComplexConjugate over a spinor

### DISCUSSION THREAD

**Discussion by smhrobi** [FeynCalc/feyncalc#114] [Q&A]
Hi,

I am trying to calculate a test amplitude using FeynCalc 9.3.1. My mathematica version is 12.1.

While trying to take the complex conjugate I face the following issue:
.........................................................................................................................................................................................
In[70]:= Spinor[Momentum[(p1 + p2), D], MU, 1] // ComplexConjugate

During evaluation of In[70]:= FermionSpinSum::failmsg: Error! FermionSpinSum encountered a fatal problem and must abort the computation. The problem reads: The input contains Spinor objects with incorrect syntax.

Out[70]= $Aborted
.........................................................................................................................................................................................

But if I put, for example, Spinor[Momentum[(p1), D], MU, 1] , it works. This was not an issue in version 9.3.0.

Thanks

Syed

**Comment by vsht** [comment]
Hi,

that's right. This is because of #76. 

The point is that the quantity inside Momentum of a Spinor should
be a single symbol (e.g. p1). Notice that the sign also matters a lot, because this is how FeynCalc distinguishes
between `u` and `v` spinors. For this reason I usually use `FCReplaceMomenta` for replacing momenta instead of
naive replacement rules:  `FCReplaceMomenta` doesn't touch the momenta of spinors and polarization vectors.

So having something like `Spinor[Momentum[(p1 + p2), D], MU, 1]` is like asking for troubles, since at some point
the spin sum will be evaluated incorrectly. The warning message is meant to protect you for that at an early stage.

Cheers,
Vladyslav

**Reply by smhrobi** [reply]
Originally I assigned p4=p1+p2-p3 beforehand for a 2->2 process and that was creating the issue. Thanks for the clarification.

**Reply by vsht** [reply]
I would keep the original momenta up to the point when you are done with the Dirac algebra.
The critical part here is clearly  the application of the Dirac equation. You usually want to use
at an earlier stage to simplify as many things as possible.

Here you can use [FCRerouteMomenta](https://feyncalc.github.io/FeynCalcBook/ref/FCRerouteMomenta.html) to make the amplitudes more compact without loss of generality. Essentially, it just improves the routing as compared to what you get e.g. from FeynArts.

Then  you can carry out your replacements via FCReplaceMomenta and see what is left. Of course in
```
FCReplaceMomenta[GS[p4] . SpinorU[p4,m],{p4->p1+p2-p3}]
```
you cannot apply the Dirac equation anymore after having eliminated p4. So depending on what you are calculating
(amplitude or amplitude squared), you might need to take care of that at a later point. For the squared amplitude
it shouldn't really matter, though.
